metrics/firebase/functions/index.js

```diff
@@ -72,12 +77,37 @@ exports.seedData = functions.https.onRequest(async (req, resp) => {
 }
 );
 
-/// Creates random Date in the range from start to end.
+/**
+ * Checks whether the email domain is in the allowed domains list.
+ *
+ * @param {Object} data - a request object that accepts such fields:
+ * - `emailDomain` - the email domain to validate.
+ *
+ * @return {Object} containing such fields:
+ * - `isValid` - indicates whether the email domain from the request is a valid or not.
+ */
+exports.validateEmailDomain = functions.https.onCall(async (data, context) => {
+    let requestData = data || {};
+
+    let userEmailDomain = requestData.emailDomain || '';
+    let allowedDomainsSnapshot = await admin.firestore().collection('allowed_email_domains').get();
```

Comment:
Can we query for a specific domain that we received from the user, rather than getting all of them to do filtering manually?